sickle/app.py

```diff
@@ -104,9 +107,20 @@ def __init__(self, endpoint,
         else:
             raise TypeError(
                 "Argument 'iterator' must be subclass of %s" % BaseOAIIterator.__name__)
-        self.max_retries = max_retries
-        self.retry_status_codes = retry_status_codes or [503]
-        self.default_retry_after = default_retry_after
+
+        if default_retry_after is not None:
+            logger.warning("default_retry_after is no longer supported, please use retry_backoff_factor instead.")
+
+        retry_adapter = requests.adapters.HTTPAdapter(max_retries=Retry(
+            total=max_retries,
+            backoff_factor=retry_backoff_factor,
+            status_forcelist=retry_status_codes or [429, 500, 502, 503, 504],
+            method_whitelist=frozenset(['GET', 'POST'])
```

Comment:
Note: `method_whitelist` has been [**deprecated** in urllib3 v1.26](https://urllib3.readthedocs.io/en/stable/changelog.html#id31) and [**removed** in its v2](https://urllib3.readthedocs.io/en/stable/changelog.html#id10). And since requests pins urllib3 to a [range](https://github.com/psf/requests/blob/72eccc8dd8b7c272e520f22b0256386c80864e94/setup.py#L64) which includes the latter, IMHO the newer idiom must be used here:

```suggestion
            allowed_methods=frozenset(['GET', 'POST'])
```

Otherwise, you'll get a 
```
    retry_adapter = requests.adapters.HTTPAdapter(max_retries=Retry(
TypeError: __init__() got an unexpected keyword argument 'method_whitelist'
```